src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import topology.opens
+import ring_theory.ideal_operations
+
+/-!
+# Prime spectrum of a commutative ring
+
+The prime spectrum of a commutative ring is the type of all prime ideals.
+It is naturally endowed with a topology: the Zariski topology.
+
+(It is also naturally endowed with a sheaf of rings,
+but that sheaf is not constructed in this file.
+It should be contributed to mathlib in future work.)
+
+## Inspiration/contributors
+
+The contents of this file draw inspiration from
+https://github.com/ramonfmir/lean-scheme
+which has contributions from Ramon Fernandez Mir, Kevin Buzzard, Kenny Lau,
+and Chris Hughes (on an earlier repository).
+
+-/
+
+universe variables u v
+
+variables (R : Type u) [comm_ring R]
+
+/-- The prime spectrum of a commutative ring `R`
+is the type of all prime ideal of `R`.
+
+It is naturally endowed with a topology (the Zariski topology),
+and a sheaf of commutative rings (not yet in mathlib).
+It is a fundamental building block in algebraic geometry. -/
+def prime_spectrum := {I : ideal R // I.is_prime}
+
+variable {R}
+
+namespace prime_spectrum
+
+/-- A method to view a point in the prime spectrum of a commutative ring
+as an ideal of that ring. -/
+abbreviation as_ideal (x : prime_spectrum R) : ideal R := x.val
+
+instance as_ideal.is_prime (x : prime_spectrum R) :
+  x.as_ideal.is_prime := x.2
+
+@[ext] lemma ext {x y : prime_spectrum R} :
+  x = y ↔ x.as_ideal = y.as_ideal :=
+subtype.ext
+
+/-- The zero locus of a set `I` of elements of a commutative ring `R`
```

Comment:
Is that a completely standard notation to use `I` for a set of elements in the ring? I fear it might be confused for an ideal because of the letter `I`.

Reply:
I chose `I`, because in practice you usually use ideals. But the definition works for a general set. So this is meant to be the "ideal-side"  of the story. If you think it's too confusing, I can change it.

Reply:
Yes, I think I'd rather see `s` here.